File: client/src/common/model/classModels/subject_model.ts
export class SubjectModel {
    public id: string;
    public title: string;
    public author: string;
    public enable: boolean;
    public classId: string;

    constructor(
        id: string,
        title: string,
        author: string,
        enable: boolean,
        classId: string
    ) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.enable = enable;
        this.classId = classId;
    }
}


export class SubjectModelWithOutId {
    public title: string;
    public author: string;
    public enable: boolean;
    public classId: string;

    constructor(
        title: string,
        author: string,
        enable: boolean,
        classId: string
    ) {

        this.title = title;
        this.author = author;
        this.enable = enable;
        this.classId = classId;
    }
}




export class ProceedExam {
    public subjectId: string
    public studentId: string

    constructor({ subId, studId }: { subId: string, studId: string }) {
        this.studentId = studId
        this.subjectId = subId
    }
}

File: server/app/repo/models/subject/subject_model.py
from ...dependecy import Base
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import UUID, String, Boolean, DateTime, ForeignKey
import uuid
from datetime import datetime

class SubJectModel(Base):
    __tablename__ = "subjects"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title: Mapped[str] = mapped_column(String(100), index=True)
    author: Mapped[str] = mapped_column(String(100), index=True)
    enable: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.now)

    class_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("classes.id", ondelete="CASCADE"),
        nullable=False
    )

    class_: Mapped["ClassModel"] = relationship("ClassModel", back_populates="subjects")


    question: Mapped[list["QuestionModel"]] = relationship(
        "QuestionModel",
        back_populates="subject",
        cascade="all, delete-orphan"
    )

    # One-to-one with TimerModel
    timer: Mapped["TimerModel"] = relationship(
        "TimerModel",
        back_populates="subject",
        uselist=False,
        cascade="all, delete-orphan"
    )


    filter_question: Mapped["FilterQuestionModel"] = relationship(
        "FilterQuestionModel",
        back_populates="subject",
        uselist=False,
        cascade="all, delete-orphan"
    )
    
    
    scores: Mapped[list["StudentScoreModel"]] = relationship(
        "StudentScoreModel",
        back_populates="subject",
        cascade="all, delete-orphan"
    )
